src/software/backend/input/replay_logging/replay_logger.cpp

```diff
@@ -0,0 +1,84 @@
+#include "software/backend/input/replay_logging/replay_logger.h"
+
+#include <google/protobuf/util/delimited_message_util.h>
+
+#include <fstream>
+
+#include "software/logger/logger.h"
+
+namespace fs = std::experimental::filesystem;
+
+ReplayLogger::ReplayLogger(const std::string& out_dir_path, int _msgs_per_chunk)
+    : OrderedThreadedObserver<SensorMsg>(2000),
+      current_chunk(),
+      current_chunk_idx(0),
+      output_dir_path(out_dir_path),
+      msgs_per_chunk(_msgs_per_chunk)
+{
+    // check if directory exists, if not make a directory
+    if (fs::exists(output_dir_path))
+    {
+        if (!fs::is_directory(output_dir_path))
+        {
+            throw std::invalid_argument(out_dir_path + " exists and is not a directory!");
+        }
+        if (!fs::is_empty(output_dir_path))
+        {
+            // this is better behavior than either adding more chunks to the same
+            // directory (and having one directory end up with multiple replays) or
+            // silently overwriting and destroying a previous replay_logging
+            throw std::invalid_argument(out_dir_path +
+                                        " is not empty! Find another directory!");
+        }
+    }
+
+
+    if (!fs::exists(output_dir_path))
+    {
+        fs::create_directory(output_dir_path);
+        LOG(INFO) << "Created directory " << output_dir_path;
+    }
+
+    LOG(INFO) << "Logging to " << output_dir_path.string();
+}
+
+
+ReplayLogger::~ReplayLogger()
+{
+    saveCurrentChunk();
+}
+
+
+void ReplayLogger::onValueReceived(SensorMsg msg)
+{
+    if (msg.has_ssl_vision_msg())
+    {
+        LOG(INFO) << "Logging vision frame with t_sent="
+                  << msg.ssl_vision_msg().detection().t_sent();
+    }
```

Comment:
I think this LOG will likely clog up the output quite a lot. Perhaps let's just log when we transition chunks?

@jonathanlew @akhilveeraghanta @MathewMacDougall for your opinions

Reply:
Or we can make this a DEBUG level log?

Reply:
I'd be fine with that.

Reply:
Can we just log at the beginning saying logging is turned on?

Reply:
this was supposed to be deleted, will log when we transition chunks